include/deal.II/numerics/vector_tools.h

```diff
@@ -924,6 +928,26 @@ namespace VectorTools
                 VectorType &vec_result);
 
   /**
+   * Generic implementation for the project() function on parallel Triangulations with
+   * FiniteElements that are supported by the MatrixFree class.
+   * Use this function if you use more than four components or the degree of your FiniteElement
+   * is higher than eight. You have to specify the first template arguments yourself.
+   * @p vec_result is expected to not have any ghost entries.
+   */
+  template <int components, int fe_degree, int dim, typename VectorType, int spacedim>
+  void project_generic (const Mapping<dim, spacedim>                              &mapping,
```

Comment:
do we want to have this function? I thought interface of `VectorTools` does not change at all, with a few exceptions of RHS calculations. So my understanding is that this should go to `.templates.h`.